vbfml/scripts/err_analysis

```diff
@@ -455,5 +492,292 @@ def compare_hist(ctx, input_dirs: str, number_epochs: int, save_name: str) -> No
     plt.savefig(pjoin(outdir, f"comparison_val_loss_{save_name}.pdf"))
 
 
+@cli.command()
+@click.pass_context
+@click.option(
+    "-i",
+    "--input-dir",
+    required=True,
+    help="path to the prediction directory of a model",
+)
+def roc(ctx, input_dir: str) -> None:
+    """
+    compute the roc curve of the model.
+    only compute and write in pkl file so it allows to use plot-roc on several model to compare in in one plot
+    """
+
+    # load model and prediction pickle file
+    loader = TrainingLoader(os.path.dirname(input_dir.rstrip("/")))
+    model = loader.get_model()
+    features, predictions, label_encoding, labels = load_cache(input_dir)
+    mjj = features["mjj"]
+    detajj = features["detajj"]
+
+    # compute ROC and precision recall curve
+    precision, recall, thresholds_prc = precision_recall_curve(labels, predictions)
+    fpr, tpr, thresholds_roc = roc_curve(labels, predictions)
+
+    # do the same for a simple cut on mjj and detajj
+    mjj_norm = mjj / mjj.max()
+    detajj_norm = detajj / detajj.max()
+    fpr_m, tpr_m, thresh_roc_m = roc_curve(labels, mjj_norm)
+    fpr_eta, tpr_eta, thresh_roc_eta = roc_curve(labels, detajj_norm)
+
+    output_pkl = pjoin(input_dir, "roc.pkl")
+    # Mention the output file path in the logging
+    print(f"Writing output to: {output_pkl}")
+
+    data = {
+        "fpr": fpr,
+        "tpr": tpr,
+        "thresh_roc": thresholds_roc,
+        "precision": precision,
+        "recall": recall,
+        "thresholds_prc": thresholds_prc,
+        "fpr_m": fpr_m,
+        "tpr_m": tpr_m,
+        "mjj_max": mjj.max(),
+        "thresh_roc_m": thresh_roc_m,
+        "fpr_eta": fpr_eta,
+        "tpr_eta": tpr_eta,
+        "thresh_roc_eta": thresh_roc_eta,
+        "eta_max": detajj.max(),
```

Comment:
Just to confirm, I had a question about `mjj_max` and `eta_max` here. Is the `.max()` operation done on `mjj` and `detajj` arrays to flatten it into a 1D array at the end, or is there some other purpose?

Also, can we update some variable namings here for clarity? I have the following suggestions, but let me know what you think:
- When you use `eta`, it really means `detajj` right? So I would replace the use of `eta` with `detajj` for clarity.
- If the `.max()` operation is being done to flatten the array, I think we can remove the `_max` postfix from `mjj` and `detajj` in the cache. So you can simply save them as `mjj` and `detajj`.
- Finally, I think you can do something like `fpr_m` -> `fpr_mjj` and `fpr_eta` -> `fpr_detajj` (also for tpr)